src/proxy.rs

```diff
@@ -502,11 +527,129 @@ fn handle_server_event(
             );
 
             // TODO(#145): Fix unwrap
-            let handle = client_to_proxy.idle_reject(status).unwrap();
+            let handle = client_to_proxy.state.idle_reject(status).unwrap();
             trace!(role = "p2c", ?handle, "idle_reject");
         }
         client::Event::IdleDoneSent { handle } => {
             trace!(role = "p2s", ?handle, "--->");
         }
     }
 }
+
+struct ImapState<S> {
+    input_role: &'static str,
+    input_direction: &'static str,
+    input_color: Option<Color>,
+    input_fragmentizer: Fragmentizer,
+    output_role: &'static str,
+    output_direction: &'static str,
+    output_color: Option<Color>,
+    output_fragmentizer: Fragmentizer,
+    state: S,
+}
+
+impl ImapState<Client> {
+    pub fn client(state: Client) -> Self {
+        Self {
+            input_role: "s2p",
+            input_direction: "<--|",
+            input_color: Some(Color::Blue),
+            input_fragmentizer: Fragmentizer::without_max_message_size(),
+            output_role: "p2s",
+            output_direction: "--->",
+            output_color: None,
+            output_fragmentizer: Fragmentizer::without_max_message_size(),
+            state,
+        }
+    }
+}
+
+impl ImapState<Server> {
+    pub fn server(state: Server) -> Self {
+        Self {
+            input_role: "c2p",
+            input_direction: "|-->",
+            input_color: Some(Color::Red),
+            input_fragmentizer: Fragmentizer::without_max_message_size(),
```

Comment:
We have a message limit of 100 MiB on the proxy by default (because we use imap-next's [`client::Option::default()`](https://github.com/duesee/imap-next/blob/29072279a68c299d679c5712ec03c69fceff801f/src/client.rs#L44)).

Now... it's not clear anymore we stick to the limit. (I think we don't but maybe there is an invariant I'm missing.)

Reply:
No, you're right. We should use the same limit.